cnew.py

```diff
@@ -124,6 +126,9 @@ def mainfunc(username, password, chatroom, servaddr):
         else:
             if chekroom(chatroom) is True:
                 print_formatted_text(HTML("[" + obtntime() + "] " + "SNCTRYZERO > <green>A valid chatroom identity was entered</green>"))
+            elif not chatroom.isupper():
+                print_formatted_text(HTML("[" + obtntime() + "] " + "SNCTRYZERO > <green>An invalid chatroom identity (with lowercase character) was entered</green>"))
+                sys.exit()
```

Comment:
I think this might be a bit too stringent on the users joining the chatroom. You can, of course, convert the lowercase parts of the string to uppercase and then check the hexadecimal and length compliance.

Reply:
OK sure..I'll be back with the changes.